sunpy/io/_file_tools.py

```diff
@@ -57,6 +57,30 @@ def __getitem__(self, key):
     'ana': ana
 })
 
+def get_open_file(filepath, mode="rb", **kwargs):
+    """
+    Ensures the filepath is converted to an fsspec.OpenFile object.
+
+    Parameters
+    ----------
+    filepath : str, pathlib.Path, or fsspec.OpenFile
+        The file path or file object to be validated.
+    **kwargs : dict
+        Additional keyword arguments for fsspec.open.
+
+    Returns
+    -------
+    fsspec.OpenFile
+        An open file object.
+
+    Raises
+    ------
+    TypeError
+        If the filepath is not of type str, pathlib.Path, or fsspec.OpenFile.
+    """
+    if isinstance(filepath, fsspec.core.OpenFile):
+        return filepath
+    return fsspec.open(filepath, mode=mode, **kwargs)
```

Comment:
Does this function need to exist? I don't see why it's useful from this PR alone. What will the goal be?